server/backend/database/mongo/client.go

```diff
@@ -761,6 +761,57 @@ func (c *Client) CreateChangeInfos(
 	return nil
 }
 
+// DeleteOldChangeInfos delete old change infos before checkpoint to save storage
+func (c *Client) DeleteOldChangeInfos(
+	ctx context.Context,
+	docInfo *database.DocInfo,
+) error {
+	encodedDocID, err := encodeID(docInfo.ID)
+	if err != nil {
+		return err
+	}
+
+	// 01. Find the smallest serverSeq recorded in colSyncedSeqs.
+	// This is to avoid the error that the change could not be found when doing UpdateAndFindMinSyncedTicket.
+	result := c.collection(colSyncedSeqs).FindOne(
+		ctx,
+		bson.M{
+			"doc_id": encodedDocID,
+		},
+		options.FindOne().SetSort(bson.M{"server_seq": 1}),
+	)
```

Comment:
We need to compare the order of changes with `lamport` and `actor_id`(logical clock), not `server_seq`, which is a physical storage order in the server.

How about extracting the below codes with a function and use it together?
https://github.com/yorkie-team/yorkie/blob/4b333b949e03ae3f60fdbeb406a45c0c1a415541/server/backend/database/mongo/client.go#L904-L910

In fact, We're using Lamport timestamp, so we don't know whether the two changes are causal relations or concurrent relations. It would be better to deal with the problem about the relation of change in another issue later.